horizons/gui/widgets/minimap.py

```diff
@@ -465,90 +534,34 @@ def cleanup():
-		pixel_per_coord_x, pixel_per_coord_y = self._world_to_minimap_ratio
-
-		# calculate values here so we don't have to do it in the loop
-		pixel_per_coord_x_half_as_int = int(pixel_per_coord_x/2)
-		pixel_per_coord_y_half_as_int = int(pixel_per_coord_y/2)
-
-		world_min_x = self.world.min_x
-		world_min_y = self.world.min_y
-		island_col = self.COLORS["island"]
-		water_col = self.COLORS["water"]
 		location_left = self.location.left
 		location_top = self.location.top
-		if dump_data:
-			data = []
-			draw_point = lambda name, fife_point, r, g, b : data.append((fife_point.x, fife_point.y, r, g, b))
-		else:
-			draw_point = rt.addPoint
+		draw_point = rt.addPoint
 		fife_point = fife.Point(0, 0)
-
 		use_rotation = self._get_rotation_setting()
-		full_map = self.world.full_map
-
-		# loop through map coordinates, assuming (0, 0) is the origin of the minimap
-		# this facilitates calculating the real world coords
-		for x in xrange(where.left-self.location.left, where.left+where.width-self.location.left):
-			for y in xrange(where.top-self.location.top, where.top+where.height-self.location.top):
-
-				"""
-				This code should be here, but since python can't do inlining, we have to inline
-				ourselves for performance reasons
-				covered_area = Rect.init_from_topleft_and_size(
-				  int(x * pixel_per_coord_x)+world_min_x,
-				  int(y * pixel_per_coord_y)+world_min_y),
-				  int(pixel_per_coord_x), int(pixel_per_coord_y))
-				real_map_point = covered_area.center
-				"""
-				# use center of the rect that the pixel covers
-				real_map_x = int(x * pixel_per_coord_x) + world_min_x + pixel_per_coord_x_half_as_int
-				real_map_y = int(y * pixel_per_coord_y) + world_min_y + pixel_per_coord_y_half_as_int
-				real_map_coords = (real_map_x, real_map_y)
-
-				# check what's at the covered_area
-				if real_map_coords in full_map:
-					# this pixel is an island
-					tile = full_map[real_map_coords]
-					settlement = tile.settlement
-					if settlement is None:
-						# island without settlement
-						if tile.id <= 0:
-							color = water_col
-						else:
-							color = island_col
-					else:
-						# pixel belongs to a player
-						color = settlement.owner.color.to_tuple()
-				else:
-					color = water_col
-
-				if use_rotation:
-					# inlined _get_rotated_coords
-					rot_x, rot_y = self._rotate((location_left + x, location_top + y), self._rotations)
-					fife_point.set(rot_x - location_left, rot_y - location_top)
-				else:
-					fife_point.set(x, y)
-
-				draw_point(render_name, fife_point, *color)
-
-		if dump_data:
-			return data
 
+		for (x, y), color in iter_minimap_points(self.location, self.world,
+						self.COLORS["island"], self.COLORS["water"], where):
+			if use_rotation:
+				# inlined _get_rotated_coords
+				rot_x, rot_y = self._rotate((location_left + x, location_top + y), self._rotations)
+				fife_point.set(rot_x - location_left, rot_y - location_top)
+			else:
+				fife_point.set(x, y)
+			fife_point.set(x, y)
```

Comment:
Yeah, that seems totally wrong. It would essentially cancel out rotation, thus making it the way we discussed on IRC we might want to make it.